test/opf.jl

```diff
@@ -235,8 +235,8 @@ end
 
         @test result["status"] == :LocalOptimal
         @test isapprox(result["objective"], 11567.1; atol = 1e0)
-        @test isapprox(result["solution"]["bus"]["1"]["va"], 0.0; atol = 1e-4)
-        @test isapprox(result["solution"]["bus"]["4"]["va"], 0.0; atol = 1e-4)
+        @test isapprox(result["solution"]["bus"]["1"]["va"], 0.05285; atol = 1e-4)
+        @test isapprox(result["solution"]["bus"]["4"]["va"], 0.03252; atol = 1e-4)
     end
```

Comment:
This is highly suspicious.   Bus 1 and 4 are references buses.  va should be fixed to 0 by https://github.com/lanl-ansi/PowerModels.jl/blob/master/src/prob/opf.jl#L30.

Reply:
Let me try to figure out where the bug is. The thing is it is not clear to me as to where to start debugging. The convex hull formulation is correct; I have verified it multiple times. My suspicion is with the other parts of the qc code. Do you have a suggestion as to where to start with?

Reply:
Okay, I figured it out. All the QC constraints need to be templated. I shall work on getting that done.

Reply:
Good catch.  You should revisit the 24-bus test once that is done, probably it will work :-)